segtypes/n64/c.py

```diff
@@ -71,21 +71,24 @@ def split(self, rom_bytes: bytes):
             asm_out_dir = options.get_asm_path() / "nonmatchings" / self.dir
             asm_out_dir.mkdir(parents=True, exist_ok=True)
 
+            is_new_c_file = False
+
+            c_path = self.out_path()
+            if c_path:
+                if not os.path.exists(c_path) and options.get("create_new_c_files", True):
+                    self.create_c_file(self.funcs_text, asm_out_dir, c_path)
+                    is_new_c_file = True
+
             for func in self.funcs_text:
                 func_name = self.parent.get_symbol(func, type="func", local_only=True).name
 
                 if options.get_compiler() == "GCC":
                     if func_name not in self.defined_funcs:
                         self.create_c_asm_file(self.funcs_text, func, asm_out_dir, func_name)
```

Comment:
Looks good, but could you add the `or` for the GCC case too? Or is it redundant for that case?

Reply:
the GCC compiler if branch relies on the function not being in the `self.defined_funcs` list rather than the non GCC branch which checks against the `self.global_asm_funcs` list, so no need for the `or`.